src/tasks/LibraryBuilder/Templates/autoinit.c

```diff
@@ -105,11 +134,18 @@ free_aot_data (MonoAssembly *assembly, int size, void *user_data, void *handle)
 {
     munmap (handle, size);
 }
-#endif
+#endif // USES_AOT_DATA
 
 static void
 runtime_init_callback ()
 {
+    register_aot_modules ();
+
+#if defined(BUNDLED_ASSEMBLIES)
+    mono_register_assemblies_bundle ();
+    mono_register_bundle ();
```

Comment:
Why doesn't the generated method do all work needed, no real need for the mono_register_bundle call to happen here, simpler if the generated code would make the call as part of register all bundled resources.

Reply:
Placed the bundle registering function `mono_register_bundled_resources` in the generated file, as it will register both assembly bundles and satellite assembly bundles (and more if needed) as long as there were assemblies/satellite assemblies added.